group-coordinator/src/main/java/org/apache/kafka/coordinator/group/streams/topics/ConfiguredSubtopology.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group.streams.topics;
+
+import org.apache.kafka.common.message.StreamsGroupDescribeResponseData;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Internal representation of a subtopology.
+ * <p>
+ * The subtopology is configured according to the number of partitions available in the source topics. It has regular expressions already
+ * resolved and defined exactly the information that is being used by streams groups assignment reconciliation.
+ * <p>
+ * Configured subtopologies may be recreated every time the input topics used by the subtopology are modified.
+ */
+public record ConfiguredSubtopology(Set<String> repartitionSinkTopics,
+                                    Set<String> sourceTopics,
+                                    Map<String, ConfiguredInternalTopic> repartitionSourceTopics,
+                                    Map<String, ConfiguredInternalTopic> stateChangelogTopics) {
```

Comment:
nit:
Also here I would prefer a different order. First source, then sink, finally changelog.
```suggestion
public record ConfiguredSubtopology(Set<String> sourceTopics,
                                    Map<String, ConfiguredInternalTopic> repartitionSourceTopics,
                                    Set<String> repartitionSinkTopics,
                                    Map<String, ConfiguredInternalTopic> stateChangelogTopics) {
```

Reply:
Done